### PR TITLE
Create (a file) and Remove (unlink or rmdir)

### PR DESCRIPTION
**Pull request by mberhault**

### CODE REVIEW

File: filesystem/fs.go [hunk 1]
```diff
@@ -80,13 +81,59 @@ INSERT INTO fs.namespace VALUES ($3, $4, $1);
...
+		if count != 0 {
+			_ = tx.Rollback()
+			return fuse.Errno(syscall.ENOTEMPTY)
+		}
+	}
+
+	// Delete all entries.
+	const sql = `
+DELETE FROM fs.namespace WHERE (parentID, name) = ($1, $2);
+DELETE from fs.inode WHERE id = $3;
```
Comment (petermattis) [inline filesystem/fs.go]: `s/from/FROM/g`

File: filesystem/node.go [hunk 2]
```diff
@@ -74,9 +77,9 @@ func (n Node) Getattr(ctx context.Context, _ *fuse.GetattrRequest, resp *fuse.Ge
 // the directory, Lookup should return ENOENT.
 //
 // Lookup need not to handle the names "." and "..".
-func (n Node) Lookup(ctx context.Context, name string) (fs.Node, error) {
+func (n Node) Lookup(_ context.Context, name string) (fs.Node, error) {
 	if !n.IsDir {
-		return nil, fuse.ENOSYS
+		panic(fmt.Sprintf("Lookup request called on file node: %+v", n))
```
Comment (petermattis) [inline filesystem/node.go]: Is there no error you can return here? A panic seems dramatic. I would think `syscall.ENOTDIR` would be appropriate.

File: filesystem/node.go [hunk 3]
```diff
@@ -95,14 +98,24 @@ func (n Node) Lookup(ctx context.Context, name string) (fs.Node, error) {
 }
 
 // ReadDirAll returns the list of child inodes.
-func (n Node) ReadDirAll(ctx context.Context) ([]fuse.Dirent, error) {
+func (n Node) ReadDirAll(_ context.Context) ([]fuse.Dirent, error) {
+	if !n.IsDir {
+		panic(fmt.Sprintf("ReadDirAll request called on file node: %+v", n))
```
Comment (petermattis) [inline filesystem/node.go]: Ditto about error.

**Comment by petermattis** [comment]
LGTM modulo the panics.

File: filesystem/fs.go (hunk 1 above)
Comment (mberhault) [inline filesystem/fs.go]: Done.

File: filesystem/node.go (hunk 2 above)
Comment (mberhault) [inline filesystem/node.go]: Done everywhere. Either `ENOTDIR` or `EISDIR`.

**Pull request merged**